backend/src/services/compras/index.js

```diff
@@ -49,27 +57,85 @@ const updateCompra = (request, response) => {
             .send({ message: 'Missing path param id.' });
     }
 
-    // TODO: findById from DB
-    const existentCompra = listCompras().find(c => c.id == compraId)
-    if (!existentCompra) {
-        return response
-            .status(404)
-            .send({ message: `Compra ${compraId} not found.` });
-    }
-
+    const existentCompra = getCompraById(compraId, response);
     const updatedCompra = request.body;
-    console.log('updatedCompra', updatedCompra);
 
     for (let property in updatedCompra) {
         existentCompra[property] = updatedCompra[property];
     }
 
     // TODO: update on DB
-    // compras.push(existentCompra);
 
     return response.status(200).json(updatedCompra);
 }
 
+const assignCompra = (request, response) => {
+    const compraId = request.params.id;
+    let compra = getCompraById(compraId, response);
+
+    if (compra.status != status.NEW) {
+        return response
+            .status(409)
+            .send({ message: `Compra ${compraId} cannot be assigned due to its current status.` });
+    }
+
+    const assigneeId = 'XPTO'; // TODO: Define how to get userId here.
```

Comment:
Boa! Logo vou criar um método no user service pra pegar esse id

Reply:
Thank you buddy